scheduler/testing.go

```diff
@@ -262,7 +262,19 @@ func (h *Harness) Snapshot() State {
 // a snapshot of current state using the harness for planning.
 func (h *Harness) Scheduler(factory Factory) Scheduler {
 	logger := testlog.HCLogger(h.t)
-	return factory(logger, h.Snapshot(), h)
+	eventsCh := make(chan interface{})
+
+	// Listen for and log events from the scheduler.
+	go func() {
+		for e := range eventsCh {
+			switch event := e.(type) {
+			case *PortCollisionEvent:
+				h.t.Fatalf("unexpected worker eval event: %v", event.Reason)
```

Comment:
Not supposed to call Fatal in a goroutine sadly.
```suggestion
				h.t.Errorf("unexpected worker eval event: %v", event.Reason)
```